src/muya/lib/contentState/inputCtrl.js

```diff
@@ -153,6 +151,17 @@ const inputCtrl = ContentState => {
         } else {
           this.removeBlocks(startBlock, endBlock)
         }
+      } else if (startBlock.functionType === 'paragraphContent' &&
+        start.key === end.key && oldStart.key === start.key && oldEnd.key !== end.key) {
+        // GH#2269: The end block will lose all soft-lines when removing multiple paragraphs and `oldEnd`
+        //          includes soft-line breaks. The normal text from `oldEnd` is moved into the `start`
+        //          block but the remaining soft-lines (separated by \n) not. We have to append the
+        //          remaining text (soft-lines) to the new start block.
+        const endParagraph = document.querySelector(`#${oldEnd.key}`)
+        if (endParagraph) {
+          text += '\n' + getTextContent(endParagraph, [CLASS_OR_ID.AG_MATH_RENDER, CLASS_OR_ID.AG_RUBY_RENDER])
+        }
```

Comment:
It is not recommended to use too many DOM APIs(querySelector), which will cause performance penalties. how about use `endBlock` like this:

```js
if (endBlock && endBlock.text.indexOf('\n')) {
    text += endBlock.text.replace(/^.*(?=\n)/, '')
}
```

Reply:
I think this is a better solution but I need to test this because a new line is used in other content like list items too.

Reply:
ok

Reply:
Should now work as expected but I detected other issues in Muya.